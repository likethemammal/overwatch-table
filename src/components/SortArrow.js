import React, { Component } from 'react'

import css from './Table.less'

import DownSVG from 'svg-icon/dist/svg/mfglabs/chevron_down.svg'
import UpSVG from 'svg-icon/dist/svg/mfglabs/chevron_up.svg'

import general from '@likethemammal/general-components'


const { SVG } = general.components

export default class SortArrow extends Component {

    render() {

        const { sortAscending, visible } = this.props

        return <div
            className={css.icon_container}
        >
            <SVG
                raw={sortAscending ? UpSVG : DownSVG}
                className={visible ? css.icon : css.icon_empty}
            />
        </div>
    }
}