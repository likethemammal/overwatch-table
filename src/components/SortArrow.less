.container {
    composes: _icon_container from '@likethemammal/overwatch-general/lib/general.css';
}

.icon_empty {
    opacity: 0;
}