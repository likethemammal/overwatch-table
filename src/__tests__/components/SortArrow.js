import React from 'react';
import {shallow} from 'enzyme';

import general from '@likethemammal/general-components'

const shared = general.__tests__._shared.tests
const { SVG } = general.components

import SortArrow from '../../../src/components/SortArrow'


const sortAscending = true
const visible = true

const props_1 = {
    sortAscending,
    visible,
}

const props_2 = {
    sortAscending: false,
    visible: false,
}

describe('SettingsSelect', () => {
    const component = shallow(
        <SortArrow
            {...props_1}
        />
    )
    const component_2 = shallow(
        <SortArrow
            {...props_2}
        />
    )

    shared.SNAPSHOTS_SHOULD_MATCH([
        component,
        component_2,
    ])

    test('if not visible, the class should equal undefined', () => {
        expect(
            component_2.find(SVG).props().className
        ).toEqual('SVG-container')
    })

    test('if visible, the class should equal empty string', () => {
        expect(
            component.find(SVG).props().className
        ).toEqual('')
    })

    test('since its a mock file, just test that SVG component receive the stub', () => {
        expect(
            component.find(SVG).props().raw
        ).toEqual('test-file-stub')
    })
})