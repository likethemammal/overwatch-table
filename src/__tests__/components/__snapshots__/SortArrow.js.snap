// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsSelect Snapshot 0 should match 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SortArrow
    sortAscending={true}
    visible={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        className=""
        raw="test-file-stub"
      />,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "",
        "raw": "test-file-stub",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          className=""
          raw="test-file-stub"
        />,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "raw": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SettingsSelect Snapshot 1 should match 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SortArrow
    sortAscending={false}
    visible={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        className="SVG-container"
        raw="test-file-stub"
      />,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "SVG-container",
        "raw": "test-file-stub",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          className="SVG-container"
          raw="test-file-stub"
        />,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "SVG-container",
          "raw": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
